DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Repositories/TrustRepository.cs

```diff
@@ -58,11 +60,43 @@ public async Task<TrustDetails> GetTrustDetailsAsync(string uid)
         return trustDetailsDto;
     }
 
+    public async Task<TrustGovernance> GetTrustGovernanceAsync(string uid)
+    {
+        var governors = await academiesDbContext.GiasGovernances
+            .Where(g => g.Uid == uid)
+            .Select(governance => new Governor(governance.Gid!, governance.Uid!, GetFullName(governance),
+                governance.Role!, governance.AppointingBody!, governance.DateOfAppointment.ParseAsNullableDate(),
+                governance.DateTermOfOfficeEndsEnded.ParseAsNullableDate(), null))
+            .ToArrayAsync();
+        var governersDto = new TrustGovernance(
+            governors.Where(governor => governor is { IsCurrentGovernor: true, HasRoleLeadership: true }).ToArray(),
+            governors.Where(governor => governor is { IsCurrentGovernor: true, HasRoleMemeber: true })
+                .ToArray(),
+            governors.Where(governor => governor is { IsCurrentGovernor: true, HasRoleTrustee: true })
+                .ToArray(),
+            governors.Where(governor => governor.IsCurrentGovernor is false).ToArray()
+        );
+        return governersDto;
+    }
+
     private async Task<string> GetRegionAndTerritoryAsync(string uid)
     {
         return await academiesDbContext.MstrTrusts
             .Where(m => m.GroupUid == uid)
             .Select(m => m.GORregion)
             .SingleOrDefaultAsync() ?? string.Empty;
     }
+
+    private static string GetFullName(GiasGovernance giasGovernance)
```

Comment:
This will force entity framework to return every field in the GIAS Governance table (you can see the actual SQL requests made by adding `  "Serilog:MinimumLevel:Override:Microsoft": "Information"` to your user secrets file)